discord/ui/select.py

```diff
@@ -340,25 +812,37 @@ def select(
         The maximum number of items that must be chosen for this select menu.
         Defaults to 1 and must be between 1 and 25.
     options: List[:class:`discord.SelectOption`]
-        A list of options that can be selected in this menu.
+        A list of options that can be selected in this menu. This can only be used with
+        :class:`Select` instances.
+    channel_types: List[:class:`~discord.ChannelType`]
+        The types of channels to show in the select menu. Defaults to all channels. This can only be used
+        with :class:`ChannelSelect` instances.
     disabled: :class:`bool`
         Whether the select is disabled or not. Defaults to ``False``.
     """
 
-    def decorator(func: ItemCallbackType[V, Select[V]]) -> ItemCallbackType[V, Select[V]]:
+    def decorator(func: ItemCallbackType[V, BaseSelectT]) -> ItemCallbackType[V, BaseSelectT]:
         if not inspect.iscoroutinefunction(func):
             raise TypeError('select function must be a coroutine function')
+        if not issubclass(cls, BaseSelect):
+            raise TypeError(
+                f'cls must be one of {", ".join(c.__name__ for c in BaseSelect.__subclasses__())} or a subclass of one of them, not {cls!r}.'
```

Comment:
`__subclasses__` is a CPython implementation detail, and is not actually documented anywhere to my knowledge so I try to avoid using it. Not to mention that join inside an f-string looks incredibly busy.

Reply:
For future reference, this one is [documented](https://docs.python.org/3/library/stdtypes.html#class.__subclasses__) but I don't think using it here is a real benefit.